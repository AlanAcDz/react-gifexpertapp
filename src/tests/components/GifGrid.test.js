import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';

jest.mock('../../hooks/useFetchGifs');

describe('<GifGrid/> tests', () => {
    const category = 'Tests';
    test('should render properly', () => {
        useFetchGifs.mockReturnValue({ data: [], loading: true });
        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot(); 
    });
    test('should show items when images load with useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://test.com',
            title: 'Text'
        }];
        useFetchGifs.mockReturnValue({ data: gifs, loading: false });
        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBeFalsy();
        expect(wrapper.find('GifGridItem')).toHaveLength(gifs.length);
    });
});
