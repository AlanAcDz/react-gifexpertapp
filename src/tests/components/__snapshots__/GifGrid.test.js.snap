// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid/> tests should render properly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Tests
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`<GifGrid/> tests should show items when images load with useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Tests
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Text"
      url="https://test.com"
    />
  </div>
</Fragment>
`;
